study-frontend/src/Components/PageSwitcher.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {useEffect, useState, useContext} from 'react'
+import '../cssFiles/pageSwitcher.css'
+import { MdChevronRight } from "react-icons/md";
+import { MdKeyboardArrowLeft } from "react-icons/md";
+import { FetchPagesCount } from '../Services/User/FetchPagesCount';
+import { Tag } from '../Types/TagInterfaces';
+import { ThemeContext } from '../Context/ThemeContext';
+
+type Props = {
+    page: number; 
+    setPage: (page: number) => void;
+    searchValue: string;
+    tag: string
+}
+
+export const PageSwitcher = ({page, setPage, searchValue, tag}: Props) => {
+    const [pages, setPages] = useState(0);
+    const [shortCut1, setShortCut1] = useState(0);
+    const [shortCut2, setShortCut2] = useState(0);
+    const [shortCut3, setShortCut3] = useState(0);
+    const {viewportWidth} = useContext(ThemeContext);
+    
+    useEffect(() => {
+        const fetchPages = async() => {
+            try{
+               
+                if(tag === ""){
+                    const subjectId = "";
+                    const response = await FetchPagesCount({searchValue, subjectId}); 
+                    setPages(response);
+                }
+                else
+                {
+                    const subjectId = tag;
+                    const response = await FetchPagesCount({searchValue, subjectId}); 
+                    setPages(response);
+                }
+               
+
+            }catch(error)
+            {
+                console.log("can't fetch the count of pages ")
+            }
+        }
+
+        fetchPages()
+    }, [searchValue])  
+    
+    
+    useEffect(()=>{
+        const changeShortCuts = () => {
+            if(pages !== 0)
+            {
+                if(page + 4 < pages){
+                    setShortCut1(page + 1);
```

Comment:
Try not to set the page functions like this short cut 1, 2, 3 you can make this logic with objects.